ci/Publish-MailMobileArtifacts.Jenkinsfile

```diff
@@ -0,0 +1,299 @@
+import groovy.transform.Field
+@Field def releaseNotes
+
+pipeline {
+	environment {
+		PATH="${env.NODE_PATH}:${env.PATH}"
+		VERSION = sh(returnStdout: true, script: "${env.NODE_PATH}/node -p -e \"require('./package.json').version\" | tr -d \"\n\"")
+		IOS_RELEASE_NOTES_PATH = "app-ios/fastlane/metadata/default/release_notes.txt"
+	}
+
+    parameters {
+        booleanParam(
+            name: 'googlePlayStore',
+            defaultValue: false,
+            description: "Uploads android artifacts (apk) to Google PlayStore as a Draft on the public track."
+        )
+        booleanParam(
+            name: 'appleAppStore',
+            defaultValue: false,
+            description: "Uploads iOS artifacts to Apple App Store as a Draft on the public track."
+        )
+        booleanParam(
+        	name: 'github',
+        	defaultValue: false,
+        	description: "Uploads android artifact (apk) to GitHub and publish release notes."
+        )
+		string(
+			name: 'appVersion',
+			defaultValue: "",
+			description: 'Which version should be published.'
+		)
+		booleanParam(
+			name: "generateReleaseNotes",
+			defaultValue: true,
+			description: "Generate Release notes for this build."
+		 )
+    }
+
+    agent {
+		label 'linux'
+	}
+
+    stages {
+    	stage("Checking params") {
+			steps {
+				script{
+					if(!params.googlePlayStore && !params.appleAppStore && !params.github) {
+						currentBuild.result = 'ABORTED'
+						error('No artifacts were selected.')
+					}
+				}
+				echo "Params OKAY"
+			}
+    	}
+		stage("Prepare Release Notes") {
+			environment {
+				VERSION = "${params.appVersion.trim() ?: env.VERSION}"
+			}
+			when {
+				expression {
+					params.generateReleaseNotes && (params.googlePlayStore || params.appleAppStore || params.github)
```

Comment:
could be just `params.generateReleaseNotes`, you already checked that at least one of them is set in the previous step